app/controllers/api/account.php

```diff
@@ -2205,8 +2195,15 @@
 
         Authorization::setRole(Role::user($profile->getId())->toString());
 
+        $newPassword = Auth::passwordHash($password, Auth::DEFAULT_ALGO, Auth::DEFAULT_ALGO_OPTIONS);
+        $historyLimit = $project->getAttribute('auths', [])['passwordHistory'] ?? 0;
+        $history = $user->getAttribute('passwordHistory', []);
+        $history[] = $newPassword;
+        $history = array_slice($history, (count($history) - $historyLimit), $historyLimit);
+
```

Comment:
There was another bug where the password history was not updated on password recovery.